rpc/transaction.go

```diff
@@ -46,8 +49,10 @@ func adaptTransaction(t TXN) (Transaction, error) {
 		json.Unmarshal(txMarshalled, &tx)
 		return tx, nil
 	default:
-		panic("not a transaction")
+		return nil, errors.New("Internal error with adaptTransaction() : unknown transaction type")
```

Comment:
few things to consider:
* golang code style suggests to start error messages with lowercase char (because it looks more cleaner when you wrap errors i.e. "outer error: internal error: deep internal error")
* can we add here more specifics what caused an error here i.e. "unknown transaction type %s"